### PR TITLE
Add workflow to cleanup after project was created from template

### PR DESCRIPTION
**Pull request by LtdSauce**
This PR was moved from the cpp_starter_project (https://github.com/cpp-best-practices/cpp_starter_project/pull/197).
This PR implements a first simple workflow to do some cleanup inspired by cpp-best-practices/cpp_boilerplate_project#5.

As https://github.com/github/feedback/discussions/5336 is not implemented by github, this PR should workaround that need for a template project. 

Goals of this PR:
- [x] cleanup build badges when project is created from template and remove the worklfow from the project
- [x] allow to make forks to contribute to the template without triggering the workflow: Use the information if the repository is a template for this
- [x] make it possible to create another template based on this template with correct build badges easily or automatically (better: don't make it impossible): Use the information if the repository is a template for this
- [x] rename the cmake project when project is created 
- [x] remove LICENSE file so new project is not by default Public Domain, and encourage user to place their own
- [x] remove funding.yml file so new project doesn't implicitly point back to @lefticus
- [x] consider a README-Template.md that is easier to regex replace from and use that to replace existing README.md This might make setup and maintenance easier?
- [x] Test: on template repo run the workflow with random variables to check it is working and just omit the push into the repository 
- [ ] Probably have the diffs of the commit checked when run on the project template (those should be deterministic)
- [x] Use repo description to fill in a placeholder inside the README template

### CODE REVIEW

**Comment by LtdSauce** [comment]
I'll close the PR while i am working on it so not every push wants to trigger the workflows

**Comment by LtdSauce** [comment]
I guess i have everything covered. I am not sure how to test if the search and replace worked out correctly,,, currently i just print the diff.

File: .github/workflows/template-janitor.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          owner: ${{ env.NEW_ORG }}
+          repo: ${{ env.NEW_PROJECT }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      
+      # Rename all cpp_starter_project occurences to current repository and remove this workflow
+      - name: Insert new org and project
+        run: |
+          # rename the CMake project to match the github project
+          sed -i "s/myproject/${{ github.event.repository.name }}/g" CMakeLists.txt
```
Comment (lefticus) [inline .github/workflows/template-janitor.yml:39]:
```suggestion
          sed -i "s/myproject/${{ github.event.repository.name }}/gi" CMakeLists.txt configured_files/config.hpp.in
```

To catch the names in the config.hpp.in example

**Review by lefticus — CHANGES_REQUESTED** [review]

**Comment by lefticus** [comment]
This is amazing. I just left one small suggestion for catching some of the C++ things and I'm also wondering if you think we can do anything with this?

https://github.com/cpp-best-practices/cpp_boilerplate_project/blob/main/CMakeLists.txt#L38-L39

I think HOMEPAGE_URL should be relatively easy to take care of. Description, I'm concerned about newlines or something else breaking CMake.

File: .github/workflows/template-janitor.yml (hunk 1 above)
Comment (LtdSauce) [reply .github/workflows/template-janitor.yml:39]: I'll add this as soon as i can

**Comment by LtdSauce** [comment]
I agree.
I would try to cut off the description on the first newline and treat it as a `short_description` and pass that to cmake.

@lefticus i wonder if it would be a good idea to run cmake configure before commiting the changes inside the workflow? That makes it more complex but would catch any problems the replace introduces to the cmake files

File: .github/workflows/template-janitor.yml (hunk 1 above)
Comment (lefticus) [reply .github/workflows/template-janitor.yml:39]: You can just hit the "commit suggestion" button if you think it would work

**Comment by lefticus** [comment]
@LtdSauce I think it would be safe to auto-fill the description inside of CMake, because it doesn't seem to be possible to create a multi-line description when using the template project

File: .github/workflows/template-janitor.yml (hunk 1 above)
Comment (LtdSauce) [reply .github/workflows/template-janitor.yml:39]: I know. But not from the Android Github App 😅

Comment (LtdSauce) [reply .github/workflows/template-janitor.yml]: Thanks for the suggestion!

**Review by lefticus — APPROVED** [review]

**Pull request merged**